design/roadmap.md

```diff
@@ -1,179 +1,71 @@
+* In addition to using the global registry, users can configure Ksonnet to use parts on the local filesystem as dependencies.
 
-**A new command (e.g. `ks registry create`) which autogenerates registry scaffolding files, may make this easier and less error prone.**
+## Default parts
 
-## Lazily download and update dependencies (`ks install`)
+* Standard Kubernetes object types (e.g. Deployments) will be wrapped as parts, and included with ksonnet by default.
+* To reduce the complexity of the full API spec, these parts will expose only the most "useful" object fields.
+* These parts will act as local dependencies and be upgraded through the registry.
 
-*Planned for 0.9.0* ([#217](https://github.com/ksonnet/ksonnet/issues/217), [#237](https://github.com/ksonnet/ksonnet/issues/237))
+## JSON and YAML support
 
-Currently, `ks init` fails when the user doesn't have an internet connection, because it relies on two external dependencies:
-* The `k.libsonnet` for the `default` environment
-* The `incubator` registry
+* Users can add JSON, YAML, or Jsonnet file into `components/` and be able to apply them with `ks apply`. Only Jsonnet files can use parameters.
+* To better facilitate migrations, users can use `ks` to convert JSON or YAML files to Jsonnet.
 
-To avoid this issue, **ksonnet needs to lazily initialize environments and registries**, rather than trying to set these up right away.
+## Apply
 
-**This plays into the idea of a `ks install` command, which will cut down on the number of version-controlled files for a given ksonnet app**. ksonnet dependencies will be `.gitignore`-ed. *Only their metadata* (e.g. versions) will be tracked.
+* Users can apply their configurations in a consistent manner. Resources that do not need to be changed will untouched during the `ks apply` process.
 
-After a ksonnet app is downloaded (e.g. via `git clone`), `ks install` can be run to download and "flesh out" the necessary environments, registries, and packages. This is similar to `npm install`.
+## Guidance for secrets management
 
-**An associated command (e.g. `ks update`) should allow users to download the latest versions of a dependency.**
+* The Ksonnet team will supply guidance on managing secrets in Ksonnet. The team is currently investigating Bitnami's [sealed secrets](https://github.com/bitnami/sealed-secrets) as an approach.
 
-## Expose "environment metadata" to prototypes
+## Refine output of `ks diff`
 
-*Planned for 0.9.0* ([#222](https://github.com/ksonnet/ksonnet/issues/222))
+* Diff will only show the values the user changed in its output.
+* Colors in diff output can be disabled.
 
-Nearly all ksonnet prototypes expect a `namespace` parameter. However, in the Jsonnet code, there is currently no way of referencing the namespace of a given ksonnet environment.
+## le code checking with `ks lint`
```

Comment:
typo: thinking through it again, "code checking" is a bit informal.

Replace with "Static analysis with `ks lint`" ?